experimental/SetPartitions/src/Util.jl

```diff
@@ -0,0 +1,108 @@
+"""
+    new_point_values(p_upper::Vector{Int}, p_lower::Vector{Int}, 
+                     q_upper::Vector{Int}, q_lower::Vector{Int})
+
+Return two vectors which are semantically identical to `q_upper` and `q_lower`
+as set-partition and have no common values with `p_upper` and `p_lower`.
+"""
+function new_point_values(p_upper::Vector{Int}, p_lower::Vector{Int}, 
+                          q_upper::Vector{Int}, q_lower::Vector{Int})
+
+    p_points = vcat(p_upper, p_lower)
+
+    if !(isempty(p_points))
+        new_id, index = findmax(p_points)
+        new_id += 1
+    else
+        new_id = 1 
+    end
+    
+    q_points = vcat(q_upper, q_lower)
+    new_ids = Dict{Int, Int}()
+
+    for n in q_points
+        new_ids[n] = new_id
+        new_id += 1
+    end
+    
+    upper = [get(new_ids, n, -1) for n in q_upper]
+    lower = [get(new_ids, n, -1) for n in q_lower]
+    
+    return (upper, lower)
+end
+
+"""
+    normal_form(p_upper::Vector{Int}, p_lower::Vector{Int})
+
+Return two vectors which are semantically identical to `p_upper` and `p_lower`
+as set-partition and are numbered from 1 to the number of subsets.
+"""
+function normal_form(p_upper::Vector{Int}, p_lower::Vector{Int})
```

Comment:
Is this a function that a user needs to call themselves? If not, I would call it `_normal_form` instead to make it clear that it is internal. 
I have seen calls to it in basically all constructors and operations, thus the question above.